### ISSUE TITLE
[no-var-requires] False Positives for Variable Module Names

### ISSUE THREAD

**Issue by manuth** [typescript-eslint/typescript-eslint#2190]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-var-requires": ["warn"]
  }
}
```

```TS
let moduleName = "my-module";
let module = require(moduleName);
```

**Expected Result**
No error should be reported as `let module = require(moduleName);` cannot be replaced by `import module = require(moduleName);` as this would cause an error.

**Actual Result**
An error is reported.

**Comment by bradzacher** [comment]
This is a correct report.
You can use `import(moduleName)` instead of `require(moduleName)`.

**Comment by manuth** [comment]
As seen in [this snippet](https://www.typescriptlang.org/play/index.html#code/DYUwLgBAtg9gJgV1AOQIZRBAvBARFATwFpZFRcBuAKAEsoAHGAJ0lKUxyZAEcEauAFIQCy8dgEoKQA), `import = require(...)` only works with a string literal.

**Comment by bradzacher** [comment]
You misunderstand me. I'm not saying use an `import = require(...)`, I'm saying use an async import `let module = await import(moduleName)`.

Here are some articles about them:

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-4.html#dynamic-import-expressions
https://mariusschulz.com/blog/dynamic-import-expressions-in-typescript
https://basarat.gitbook.io/typescript/project/dynamic-import-expressions

**Comment by manuth** [comment]
Thank you so much - I didn't know about the dynamic import syntax.
